File: backend/service-thread/models/thread.go
package models

import (
	"time"
)

type Thread struct {
	ThreadId        string  `gorm:"type:char(36);primaryKey"`
	UserId          string  `gorm:"type:char(36);not null"`
	CommunityId     *string `gorm:"type:char(36);"`
	Content         string  `gorm:"type:text;not null"`
	Category        string  `gorm:"type:varchar(100);not null"`
	IsPoll          bool    `gorm:"default:false"`
	IsPrivate       bool    `gorm:"default:false"`
	IsScheduled     bool    `gorm:"default:false"`
	ScheduledAt     *time.Time
	IsAdvertisement bool   `gorm:"default:false"`
	HasMedia        bool   `gorm:"default:false"`
	ReplyPermission string `gorm:"type:varchar(20);default:'everyone'"`
	Pinned          bool   `gorm:"default:false"`
	CreatedAt       time.Time
	UpdatedAt       time.Time
}

type Media struct {
	ThreadId  string `gorm:"type:char(36);not null;index;primaryKey"`
	MediaURL  string `gorm:"type:text;not null;primaryKey"`
	MediaType string `gorm:"type:varchar(20);not null"`
}

type PollOption struct {
	ThreadId string `gorm:"type:char(36);not null;primaryKey"`
	Option   string `gorm:"type:varchar(255);not null;primaryKey"`
}

type PollVote struct {
	ThreadId string `gorm:"type:char(36);not null;primaryKey"`
	Option   string `gorm:"type:varchar(255);not null"`
	UserId   string `gorm:"type:char(36);not null;primaryKey"`
}

type ThreadLike struct {
	UserId    string `gorm:"type:char(36);not null;index;primaryKey"`
	ThreadId  string `gorm:"type:char(36);not null;index;primaryKey"`
	CreatedAt time.Time
}

type ThreadBookmark struct {
	UserId    string `gorm:"type:char(36);not null;index;primaryKey"`
	ThreadId  string `gorm:"type:char(36);not null;index;primaryKey"`
	CreatedAt time.Time
}

type ThreadRepost struct {
	UserId    string `gorm:"type:char(36);not null;index;primaryKey"`
	ThreadId  string `gorm:"type:char(36);not null;index;primaryKey"`
	Text      string `gorm:"type:text"`
	CreatedAt time.Time
}

type ThreadReply struct {
	Id        string `gorm:"type:char(36);primaryKey"`
	ThreadId  string `gorm:"type:char(36);not null;index"`
	UserID    string `gorm:"type:char(36);not null"`
	Content   string `gorm:"type:text;not null"`
	IsPinned  bool   `gorm:"default:false"`
	CreatedAt time.Time
}

File: frontend/src/types/thread.ts
export interface Media {
    id: string;
    postId: string;
    url: string;
    type: "image" | "gif" | "video";
    timestamp: string;
}

export interface ThreadResponse {
    threads: Thread[];
}

export interface ThreadDetailResponse {
    thread: Thread;
    replies: ThreadReply[];
}

export interface Thread {
    thread_id: string;
    user_id: string;
    content: string;
    category: string;
    is_advertisement: boolean;
    reply_permission: string;
    pinned: boolean;
    media: Media[];
    poll_options: PollOption[];
    like_count: number;
    repost_count: number;
    reply_count: number;
    posted_at: string;
    is_private: boolean;
    is_liking: boolean;
    is_bookmarking: boolean;
    is_reposting: boolean;
}

export interface Media {
    id: string;
    media_url: string;
    media_type: string;
}

export interface PollOption {
    option: string;
    vote_count: number;
    is_voting: boolean;
}

export interface ThreadLike {
    id: string;
    user_id: string;
}

export interface ThreadRepost {
    id: string;
    user_id: string;
    text: string;
}

export interface ThreadReply {
    id: string;
    user_id: string;
    content: string;
    is_pinned: boolean;
    timestamp: string;
}
